qa/L0_input_validation/input_validation_test.py

```diff
@@ -0,0 +1,108 @@
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+
+sys.path.append("../common")
+
+import unittest
+
+import numpy as np
+import test_util as tu
+import tritonclient.grpc as tritongrpcclient
+from tritonclient.utils import InferenceServerException
+
+
+class InputValTest(tu.TestResultCollector):
+    def test_input_validation_required_empty(self):
+        triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+        inputs = []
+        with self.assertRaises(InferenceServerException) as e:
+            triton_client.infer(
+                model_name="input_all_required",
+                inputs=inputs,
+            )
+        err_str = str(e.exception)
+        self.assertIn(
+            "expected 3 inputs but got 0 inputs for model 'input_all_required'. Got input(s) [], but missing required input(s) ['INPUT0','INPUT1','INPUT2']. Please provide all required input(s).",
+            err_str,
+        )
+
+    def test_input_validation_optional_empty(self):
+        triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+        inputs = []
+        with self.assertRaises(InferenceServerException) as e:
+            triton_client.infer(
+                model_name="input_optional",
+                inputs=inputs,
+            )
+        err_str = str(e.exception)
+        self.assertIn(
+            "expected number of inputs between 3 and 4 but got 0 inputs for model 'input_optional'. Got input(s) [], but missing required input(s) ['INPUT0','INPUT1','INPUT2']. Please provide all required input(s).",
+            err_str,
+        )
+
+    def test_input_validation_required_missing(self):
+        triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+        inputs = []
+        inputs.append(tritongrpcclient.InferInput("INPUT0", [1], "FP32"))
+
+        inputs[0].set_data_from_numpy(np.arange(1, dtype=np.float32))
+
+        with self.assertRaises(InferenceServerException) as e:
+            triton_client.infer(
+                model_name="input_all_required",
+                inputs=inputs,
+            )
+        err_str = str(e.exception)
+        self.assertIn(
+            "expected 3 inputs but got 1 inputs for model 'input_all_required'. Got input(s) ['INPUT0'], but missing required input(s) ['INPUT1','INPUT2']. Please provide all required input(s).",
+            err_str,
+        )
+
+    def test_input_validation_optional(self):
```

Comment:
I think adding another test for a model with **only** optional inputs would be good to make sure we're not accidentally raising an error on models that don't require any inputs.